Added content_filter cog (#186)
PyDrocsid/cogs: moderation/content_filter/documentation.md

> # Content Filter

Reviewer: Please write a documentation

# Content Filter

> Contains commands to do setup checks for blacklisted expressions in every message.

Reviewer 2: "setup checks"?

> Contains commands to do set up checks for black listed expressions in every message.

Reviewer: ```suggestion
Contains commands to do setup checks for blacklisted expressions in every message.
```

!!! note

>     User with the `content_filter.bypass` permission are not effected by the check.

Reviewer 2: ```suggestion
    Users with the `content_filter.bypass` permission are not affected by these checks.
```

## `content_filter`

> Contains subcommands to do edit/show the content_filter. <br>

Reviewer 2: ```suggestion
Contains subcommands to manage the content filter.
```

```css
.[content_filter|cf]
```

Required Permissions:

- `content_filter.read`

### `add`

> This command is used to add a new expression to the filter. <br>

Reviewer 2: ```suggestion
Adds a new regular expression to the filter.
```

> This command is used to add a new expressions for the filter. <br>

Reviewer: ```suggestion
This command is used to add a new expression to the filter. <br>
```

```css
.content_filter [add|+] <regex> <delete> <description>
```

Arguments:

|   Argument    |         Required          | Description                                                                                |
|:-------------:|:-------------------------:|:-------------------------------------------------------------------------------------------|
|    `regex`    | :fontawesome-solid-check: | The [regex](https://regex101.com/) the filter should use                                   |
|   `delete`    | :fontawesome-solid-check: | True/False to indicate if the message should be deleted if a match for the regex was found |

> | `description` | :fontawesome-solid-check: | A description for the entry (shown on the list)                                            |

Reviewer 2: All text should be left aligned

Required Permissions:

- `content_filter.read`
- `content_filter.write`

> ### `list`

Reviewer 2: This command does not exist

> This commands shows a list of every blacklisted expression. <br>

Reviewer: ```suggestion
This command shows a list of every blacklisted expression. <br>
```

```css
.content_filter [list|l]
```

Required Permissions:

- `content_filter.read`

### `remove`

> This command is used to remove an expressions from the filter. <br>

Reviewer 2: ```suggestion
Removes a regular expression from the filter.
```

```css

> .content_filter [remove|-] <pattern_id>

Reviewer 2: Update aliases

```

Arguments:

> | Argument  |         Required          | Description                                     |

Reviewer 2: All text should be left aligned

|:---------:|:-------------------------:|:------------------------------------------------|

> | `pattern` | :fontawesome-solid-check: | The ID from the pattern (shown by list command) |

Reviewer 2: Update description

> | `pattern_id` | :fontawesome-solid-check: | The ID from the command (shown by list command) |

Reviewer: Is it really a command-id, when it says "pattern_id" in the argument column?

Required Permissions:

- `content_filter.read`
- `content_filter.write`

### `update`

> Contains subcommands to do edit the content_filter. <br>

Reviewer 2: ```suggestion
Contains subcommands to do edit the content filter rules.
```

```css
.content_filter [update|u]
```

Required Permissions:

- `content_filter.read`
- `content_filter.write`

#### `description`

> This command can be used to set a new description for an already existing filter.

Reviewer 2: ```suggestion
Sets a new description for an existing filter.
```

```css
.content_filter update [description|d] <pattern_id> <new_description>
```

Arguments:

|     Argument      |         Required          | Description                                     |
|:-----------------:|:-------------------------:|:------------------------------------------------|
|     `pattern`     | :fontawesome-solid-check: | The ID from the pattern (shown by list command) |

> | `new_description` | :fontawesome-solid-check: | The new description for the filter              |

Reviewer 2: All text should be left aligned

Required Permissions:

- `content_filter.read`
- `content_filter.write`

#### `regex`

> This command can be used to edit the regex for an already existing filter.

Reviewer 2: ```suggestion
Edits the regular expression of an existing filter.
```

```css
.content_filter update [regex|r] <pattern_id> <new_regex>
```

Arguments:

|  Argument   |         Required          | Description                                     |
|:-----------:|:-------------------------:|:------------------------------------------------|
|  `pattern`  | :fontawesome-solid-check: | The ID from the pattern (shown by list command) |

> | `new_regex` | :fontawesome-solid-check: | The new regex for the filter to check for       |

Reviewer 2: All text should be left aligned

Required Permissions:

- `content_filter.read`
- `content_filter.write`

> #### `toggle_delete`

Reviewer 2: Update command

This command can be used to toggle the delete-status for an already existing filter.

```css
.content_filter update [toggle_delete|td] <pattern_id>
```

Arguments:

| Argument  |         Required          | Description                                     |
|:---------:|:-------------------------:|:------------------------------------------------|

> | `pattern` | :fontawesome-solid-check: | The ID from the pattern (shown by list command) |

Reviewer 2: All text should be left aligned
